capstone/capapi/views.py

```diff
@@ -56,19 +61,118 @@ class CitationViewSet(viewsets.GenericViewSet, mixins.RetrieveModelMixin, mixins
 
 
 class CaseViewSet(viewsets.GenericViewSet, mixins.RetrieveModelMixin, mixins.ListModelMixin,):
-    """
-    Browse all cases
-    """
     serializer_class = serializers.CaseSerializer
     http_method_names = ['get']
+    lookup_field = 'slug'
     queryset = models.CaseMetadata.objects.all()
     renderer_classes = (renderers.BrowsableAPIRenderer, renderers.JSONRenderer)
-    lookup_field = 'slug'
+    permission_classes = (permissions.IsAPIUser,)
     filter_class = filters.CaseFilter
 
+    def download_many(self):
+        all_cases = self.queryset.all()
+        # get max number of cases to download from query
+        # set max number to daily max if unspecified
+        max_num = int(self.request.query_params.get('max', settings.API_CASE_DAILY_ALLOWANCE))
+
+        try:
+            to_filter = view_helpers.generate_filters_from_query_params(self.request.query_params)
+            cases = all_cases.filter(to_filter).order_by('decision_date')
+        except TypeError:
+            cases = all_cases.order_by('decision_date')
+
+        if cases.count() == 0:
+            return JsonResponse({
+                'message': 'Request did not return any results.'
+            }, status=404, )
+
+        # get page number for paginated requests
```

Comment:
Rather than roll your own pagination here, I think you can just do `cases = self.paginate_queryset(cases)`